Sources/_StringProcessing/Engine/Registers.swift

```diff
@@ -162,17 +162,21 @@ extension Processor.Registers {
   mutating func reset(_ sentinel: Input.Index) {
     // note: Is there any issue with the program transform functions holding
     // state and not getting reset here? Do we care?
-    self.bools = Array(repeating: false, count: info.bools)
-    self.ints = Array(repeating: 0, count: info.ints)
-    self.floats = Array(repeating: 0, count: info.floats)
-    self.positions = Array(repeating: sentinel, count: info.positions)
-    self.values = Array(
-      repeating: SentinelValue(), count: info.values)
+    func clear<T>(_ xs: inout [T], _ v: T) {
+      for idx in xs.indices {
+        xs[idx] = v
+      }
+    }
```

Comment:
Or as an extension on `MutableCollection`. @natecook1000 does such an algorithm exist?

Reply:
We could try these out here before adding them to swift-algorithms:

```swift
extension MutableCollection {
  mutating func setAll(to element: Element) {
    self.withEach { $0 = element }
  }
  
  mutating func withEach(_ body: (inout Element) throws -> Void) rethrows {
    var i = startIndex
    while i < endIndex {
      try body(&self[i])
      formIndex(after: &i)
    }
  }
}

// usage:
self.bools.setAll(to: false)
```

Reply:
Also considering an underscore them so that no one accidentally ships it 😅